kustomize/commands/edit/add/addreplacement.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2023 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package add
+
+import (
+	"errors"
+	"github.com/spf13/cobra"
+	"log"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/kustomize/kustomize/v5/commands/internal/kustfile"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+type addReplacementOptions struct {
+	Replacement types.ReplacementField
```

Comment:
question: are there any specific benefits from using the type `ReplacementField` here, since only `path` seems to be allowed?

Reply:
I ref ``edit add patch`` and use ``ReplacementField`` in case we need more fields in ``edit add replacement``
does this make sense?

Reply:
Yes, that's fair.